packages/nx/src/generators/tree.ts

```diff
@@ -320,6 +332,19 @@ export class FsTree implements Tree {
     }
   }
 
+  // Marks FsTree as final.
+  seal() {
+    this.enabled = false;
+  }
+
+  private checkEnabled() {
+    if (!this.enabled) {
+      throw new Error(
+        'Attempted to further modify the tree after changes were committed to disk. Using the `tree` instance is not valid after generator execution has concluded.'
+      );
```

Comment:
```suggestion
      throw new Error(
        'The tree has already been committed to disk. It can no longer be modified. Do not modify the tree during a GeneratorCallback and that Promises have resolved before the generator returns or resolves.'
      );
```